lambdas/data-migration2/src/pdrs.ts

```diff
@@ -41,7 +41,13 @@ export const migratePdrRecord = async (
   const pdrPgModel = new PdrPgModel();
   const providerPgModel = new ProviderPgModel();
 
-  const existingRecord = await pdrPgModel.get(knex, { name: dynamoRecord.pdrName });
+  let existingRecord;
+
+  try {
+    existingRecord = await pdrPgModel.get(knex, { name: dynamoRecord.pdrName });
+  } catch (error) {
+    logger.info(error);
+  }
```

Comment:
This is due to the new `BasePgModel.get()` error handling. We now throw a `RecordNotFound` if we can't find a record. Previously it was just returning undefined (or similar, either way it wasn't throwing an error).

Happy to arrange this differently but the idea here is _if we can't find an existing record, just log the error and continue. If we did find an existing record, throw the `RecordAlreadyMigrated ` error._

Reply:
Actually, we can just do the existing record check in the try block. Doing the same for data-migration2 as that test was failing in CI as well:

https://github.com/nasa/cumulus/pull/2057/commits/6a4ad3ed5599721d189437a71e69d8eb4b83daa7

Reply:
I incorrectly refactored this this morning. Pushing a change in a minute and will update this comment.

Reply:
I'm not sure I understand why letting the error be thrown from by `.get` is a problem. Throwing an error when trying to migrate any record should just mean that migrating only that record is reported as a failure and continues. 

Not adding this try/catch seems like it has little effect if the error would already have been caught and logged here?
https://github.com/nasa/cumulus/pull/2057/files#diff-d2a13da1acd9dc01f64a117c5c0ea54e97edf2eaac944e0edc634bb53ce030f8R128

Reply:
We might be getting our wires slightly crossed. I think the issue here is that `granulePgModel.get()` will actually throw an error under normal circumstances if the granule has NOT already been migrated. So if `get()` throws an error, that means that we should actually not throw it and keep going to migrate that record.

If `granulePgModel.get` DOES return a record, THEN we want to throw an error because that record has already been migrated.

Does that make sense? It's a bit confusing the way it's arranged.

Reply:
>We might be getting our wires slightly crossed. I think the issue here is that granulePgModel.get() will actually throw an error under normal circumstances if the granule has NOT already been migrated. So if get() throws an error, that means that we should actually not throw it and keep going to migrate that record.

Ah yeah, you're correct. But then I presume we need to make this change across all of our data migrations. And also the changes should only trap the RecordDoesNotExist errors and throw the others:

```suggestion
  let existingRecord;

  try {
    existingRecord = await pdrPgModel.get(knex, { name: dynamoRecord.pdrName });
  } catch (error) {
    // Swallow any non-RecordDoesNotExist errors and proceed with migration, otherwise re-throw the error
    if (!(error instanceof RecordDoesNotExist)) {
      throw error;
    }
  }
```

Reply:
👍 https://github.com/nasa/cumulus/pull/2057/commits/2af8a4f3c383a6be9720cc620264c544505cce0a